### PR TITLE
Refactor/Cleanup Presentation of Graph Task and Path Highlighting

### PR DESCRIPTION
**Pull request by ryanahamilton**
This is a collection of related updates to improve the overall user experience of the Graph view.

I tested these updates across varying DAG shapes/sizes including Task Groups. This testing also covered the hovering/clicking of the task status legend and the search filter functionality.

### General
- Simplifies JavaScript by removing the redundant inline styling for highlighting tasks and paths between them. This is accomplished by pairing a `data-highlight` attribute with CSS keyed off of it.
- I tried not to go overboard (but couldn't help myself in a few places) on the syntax cleanup as that will be handled in the migration to an external JavaScript file for #14115.
- Changes the "edge" (lines between tasks) curve to be smooth instead of angled straight lines. Single, smooth lines are easier for eyes to follow the path of than multiple jointed lines. This was accomplished by utilizing the `d3-shape` library.

| Before | After |
|---|---|
|  Image 2021-04-07 at 1 52 54 PM | Image 2021-04-07 at 1 54 31 PM  |

- Slightly increases the vertical node separation (`nodeSep`) to make the graphs feel a little less crowded. This also reduces the amount that the task nodes overlap with the edges.

| Before | After |
|---|---|
| Image 2021-04-07 at 2 11 46 PM  | Image 2021-04-07 at 2 11 15 PM  |

- Fixed an unreported bug when you hover or click on "no_status" in the status legend at the top, the correlating tasks would not highlight. 

| Before | After |
|---|---|
| Screen Recording 2021-04-07 at 02 34 48 PM  | Screen Recording 2021-04-07 at 02 35 37 PM  |

- Updated the Graph UI screenshot in the docs to reflect the changes in this PR.

### Task and Path highlighting
- Removes the varying stroke (border) weights of tasks when highlighting/hovering. This interaction isn't necessary given the non-highlighted tasks fade out. It (subjectively) makes for a smoother transition for only the stroke color to change instead of the weight as well.
- Improves the styling of the "path highlighting". Instead of styling the downstream, hovered, and upstream tasks with 3 different border colors, I've given them all the same "highlighted" color and have also given the edges that connect them a similar styling. This simplifies the pattern and visually highlights the actual path between the tasks. The upstream and downstream directions can still easily be deciphered by the edge arrows.

| Before | After |
|---|---|
|  Screen Recording 2021-04-07 at 02 47 55 PM | Screen Recording 2021-04-07 at 02 46 45 PM |

### Fixes tooltip jank

- Prevents the jumping of the tooltip position that occurs when moving your cursor within the task node
- Positions the tooltip slightly higher so it no longer overlaps with the task node's border
- Adds a very slight opacity to the tooltip background since it can cover up relevant paths between the hovered task node

| Before | After |
|---|---|
|  Screen Recording 2021-04-07 at 01 40 57 PM | Screen Recording 2021-04-07 at 01 39 06 PM  |

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: airflow/www/package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
     "bootstrap-3-typeahead": "^4.0.2",
     "codemirror": "^5.59.1",
     "d3": "^3.4.4",
+    "d3-shape": "^2.1.0",
```
Comment (kaxil) [inline airflow/www/package.json:63]: This one is a BSD license: https://github.com/d3/d3-shape/blob/master/LICENSE

Let me see if we need to add a license somewhere

Comment (kaxil) [reply airflow/www/package.json:63]: We need to include https://github.com/d3/d3-shape/blob/master/LICENSE license file in 

https://github.com/apache/airflow/tree/master/licenses and

https://github.com/apache/airflow/blob/019241be0c839ba32361679ffecd178c0506d10d/LICENSE#L246-L252

and 

https://github.com/apache/airflow/blob/019241be0c839ba32361679ffecd178c0506d10d/setup.cfg#L27-L47

**Review by kaxil — CHANGES_REQUESTED** [review]

**Review by bbovenzi — APPROVED** [review]
lgtm. This will make the js migration easier. I suspect there will be merge conflicts for #13199. But I'll handle that w/ the js migration once this is merged.

File: airflow/www/package.json (hunk 1 above)
Comment (ryanahamilton) [reply airflow/www/package.json:63]: Done!

**Review by kaxil — APPROVED** [review]

**Pull request merged**